documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+
+    * **_unauthorized_client_**
+
+      The authenticated client is not authorized to use this authorization grant type.
+
+    * **_unsupported_grant_type_**
+
+      The authorization grant type is not supported by the authorization server.
+
+    * **_invalid_scope_**
+
+      The requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner.
+
+
+* **error_description**
+
+  OPTIONAL. Human-readable ASCII text providing additional information, used to assist the client developer in understanding the error that occurred. Values for the `error_description` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+
+The parameters are included in the entity-body of the HTTP response using the `application/json` media type.
+
+`error_description` parameter is not a valid field to take decisions on the developer side, as its value may vary depending on the server implementation, language, format or context. Therefore, the developer should base their logic only on the `error` parameter.
+
+The following is a non-normative example Token Error Response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+```
+```json
+{
+  "error": "unauthorized_client"
+}
+```
+
+## The Scope Parameter
+
+Oauth scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services.
+
+The Authorization Request MUST contain the `scope` parameter, which is a space delimited, case-sensitive list of ASCII strings (scope values). Multiple scope values can be requested simultaneously, subject to the SP being registered to use those “scopes”.
+
+For OpenID requests, the scope values MUST include `openid`, to indicate that the request is an OpenID Connect request, followed by other values depending on the specific CAMARA services being requested.
+
+A purpose must be declared within the requested scope for the following grant types:
+
+* _authorization_code_
+
+* _urn:openid:params:grant-type:ciba_
+
+No purpose is required for the `client_credentials` grant type, as this grant type can only be used when no personal user data is processed. The requested scope must be set directly to the value defined for the relevant endpoint within the OAS ("YAML") specification. "Wild card" scopes (i.e. specifying only the API name) are not valid for this grant type.
+
+To declare a purpose when accessing the CAMARA APIs, it is strongly recommended to guarantee interoperability among Authorization Servers that the `scope` parameter is set to:
+
+`dpv:<dpvValue>#<technicalParameter>`
+
+* `<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+
+* `<technicalParameter>` must be either:
+
+    * one technical scope to limit the request to one API endpoint (this technical scope is described in the API spec YAML file)
```

Comment:
This restriction to allow just one technical scope should be removed from this profile.

Reply:
Yes, this was the short-term agreement reached at https://github.com/camaraproject/IdentityAndConsentManagement/issues/32#. It served to move things forward, but nobody really likes it, so we need to work in a solution for the mid-term.

In my view, this first profile should reflect the current status of the agreements, to release a 1.0 version as soon as possible and add further improvements from there, such as a better way to send the "purpose", support for PKCE (also mentioned in this issue), etc. in a 2.0 version.

Reply:
Thanks @palmerabollo for your feedback. The current short term agreements are already documented in https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.1.0/documentation/CAMARA-API-access-and-user-consent.md and are part of release 0.1 of ICM.  Release 0.2 is now focussed on urgent problems/open issues that arise due to open interpretations of release 0.1. This profile can now focus on improvements. If we are able to get this review completed early with active feedbacks & participation, we could add it as a part of release 0.2. If not, we can bring it in 0.3 release. 
I am not sure if we get any advantage by documenting this profile to reflect again the short term agreements. But we can also get more feedback from others on this.  WDYT?

Reply:
@shilpa-padgaonkar: fully agree, some text phrases above already allow for multiple scopes, so we must not limit here,

Reply:
Thanks @shilpa-padgaonkar. If I get you right, what you suggest is:

- Release 0.2 - Include a first version of this profile focused on urgent problems/open issues that aries due to open interpretations of release 0.1. _(e.g. clearly state possible error codes, required fields that weren't clear, clarify whether private_key_jwt or client_secret_basic should be used since current docs are ambiguous, etc)_
- Release 0.3 - Include further improvements in topics not yet covered _(e.g. PKCE is mentioned in some comments, adherence to FAPI 2.0, etc)._

That sounds good to me. However, I see this particular point (purposes & user consents) as a significant topic that requires some thinking to devise a better solution this time —more than just a patch on top of what we have. That is why I'd like start the discussion as soon as possible but in a separate issue, with the release 0.3 in mind.

Reply:
@palmerabollo @jpengar @diegogonmar : I went through the last TSC meeting minutes (audio recording) dated 18th Jan and @diegogonmar as a part of ICM updates has correctly mentioned that 0.2 will focus on solving open issues and 0.3 will include work on profile which will include enhancements. Are we now changing this agreement by saying profile will include contents of 0.1 and possibly 0.2?

Reply:
Hi @shilpa-padgaonkar. Activity in TSC was reporting on Identity & Consent group relevant progress, not escalation to taking any decision there. Decission in Identity&Consent was to use v0.2 for clarifications and urgent issues, I think that is clear. In parallel we decided to write OIDC profile in CAMARA. Then, once first version was proposed, as part of this github thread discussion and as per yesterday call, it was proposed to use a first version of OIDC profile as an appropriate way to cover current picture and such urgent issues targeted for v0.2. This is indeed a proposal as it could be the most effective way to solve points that need to be clarified and the urgent issues, in an organized manner. We had some interesting discussions yesterday and you weren't opposing to have profile already in v0.2, not sure if you are objecting to that now, as you sent this comment after yesterday call

Reply:
Hi @hdamker , thanks for your view. I didn't say anything about Shilpa referring to an escalation, I just added that as an information.
 
I think you say something correct: _the profile was planned for 0.3.0_. Then please read again my comment above: _as part of this github thread discussion and as per yesterday call, it was proposed to use a first version of OIDC profile as an appropriate way to cover current picture and such urgent issues targeted for v0.2_. Thus, I think its clear and I hope its fine, that this group opened a discussion on whether to include a first version of the profile for v0.2 was the best way to reflect urgent issues. 
It's not in conflict with existing agreement of having an OIDC profile for v0.3, which of course as agreed will include enhancements. So after last wednesday call you challenge if having a profile already in v0.2 it's appropriate, or at least you consider it adds no benefit. 

I'd close this part of the discussion and focus on way forward. Regarding your last paragraph, it's a valid suggestion (not saying I agree with it), can you put it in separate thread comment so we collect group opinion on that suggestion?

Reply:
@diegogonmar I agree to close this discussion and look forward. Have followed your proposal and created https://github.com/camaraproject/IdentityAndConsentManagement/pull/113#issuecomment-1940905690.